### PR TITLE
More complete doc on Analysis steps and configs to use

### PR DESCRIPTION
**Pull request by vuillaut**

### CODE REVIEW

File: docs/introduction.rst [hunk 1]
```diff
@@ -35,40 +35,104 @@ Analysis steps
 
 R1 to DL1
 ---------
-Usage of
 
-``lstchain.scripts.lstchain_data_r0_to_dl1``
+MC data
+^^^^^^^
```
Comment (chaimain) [inline docs/introduction.rst:40]: Not sure if this sub-section will be easily understood on Sphinx. Maybe just writing in a single section on both MC and real data, or listing them, might be more useful. There might be better ways to make it easily readable on Sphinx.

Maybe referencing the `lst_analysis_workflow` would be helpful to understand the sub-sections where there are distinct procedures for MC and real data, and when they are similar.

This is also applicable to the `DL2 to DL3` section

File: docs/introduction.rst [hunk 2]
```diff
@@ -35,40 +35,104 @@ Analysis steps
...
+
+Here is an example configuration file for the IRF creation step.
+
+.. toggle:: 
+
+    .. include:: examples/dl3_tool_config.json
+        :code: json
+
+
+Post DL3 analysis
```
Comment (chaimain) [inline docs/introduction.rst:134]: This has to be a major section, as we proceed towards DL4, DL5 stages

**Review by chaimain — CHANGES_REQUESTED** [review]
Some simple changes are requested.

Do we intend to use the example Jupyter notebooks on the documentation (which of course, implies that we update all the relevant notebooks!)?

File: docs/introduction.rst [hunk 3]
```diff
@@ -35,40 +35,104 @@ Analysis steps
...
+
+Here is an example configuration file for the IRF creation step.
+
+.. toggle:: 
+
+    .. include:: examples/irf_tool_config.json
+        :code: json
+
+
+Events creation
```
[the remark this answers is not in the document]
Comment (vuillaut) [reply docs/introduction.rst:116]: right

File: docs/introduction.rst (hunk 1 above)
Comment (vuillaut) [reply docs/introduction.rst:40]: Capture d’écran 2023-05-04 à 11 57 53

Comment (vuillaut) [reply docs/introduction.rst:40]: It looks ok to me this way, no?
I added a sentence with link to the workflows.

**Comment by vuillaut** [comment]
> Do we intend to use the example Jupyter notebooks on the documentation (which of course, implies that we update all the relevant notebooks!)?

That's a good idea, outside of this PR.
The notebooks could be executed and turned to HTML by the CI:
- making sure they are maintained
- serving as doc

File: docs/introduction.rst (hunk 1 above)
Comment (chaimain) [reply docs/introduction.rst:40]: Yes, it looks ok!

**Review by chaimain — APPROVED** [review]

**Pull request merged**